File: examples/primitives/python/wrappers/inheritance/rhombus_pygen.cpp
#include <pybind11/pybind11.h>
#include <pybind11/stl.h>
#include <pybind11/functional.h>

#include "cxx/inheritance/rhombus.hpp"

namespace py = pybind11;

void bindRhombus(py::module& m) {
    using namespace iegen::example;
    py::class_<iegen::example::Rhombus,iegen::example::Parallelogram> rhombus(m, "Rhombus");

    rhombus.def(py::init<double, double>(), py::arg("diagonal1"), py::arg("diagonal2"));
}

File: examples/primitives/python/src/inheritance/rhombus_pygen.py
from __future__ import annotations
from typing import *
import examples.inheritance.rhombus as pybind_rhombus_pygen
from bind_utils import *
from metaclass import *
from inheritance.parallelogram_pygen import *


class Rhombus(Parallelogram, metaclass=OriginalMethodsMetaclass):
    """"""
    
    @bind
    def __init__(self, diagonal1: float, diagonal2: float):
        pass
